src/vm/vm_memory/memory.rs

```diff
@@ -300,6 +301,28 @@ impl Memory {
     }
 }
 
+impl Display for Memory {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        writeln!(f, "memory {{")?;
```

Comment:
I dont think this is necessary, the user already knows its the memory being displayed if they print the memory